spectrocloud/resource_virtual_machine.go

```diff
@@ -113,24 +135,40 @@ func resourceVirtualMachineCreate(ctx context.Context, d *schema.ResourceData, m
 	if err != nil {
 		return diag.FromErr(err)
 	}
-	if err != nil && cluster == nil {
-		return diag.FromErr(fmt.Errorf("cluster not found: %s", clusterUid))
-	}
-	virtualMachine, err := toVirtualMachineCreateRequest(d)
-	if err != nil {
-		return diag.FromErr(err)
-	}
-	vm, err := c.CreateVirtualMachine(cluster.Metadata.UID, virtualMachine)
-	if err != nil {
-		return diag.FromErr(err)
-	}
-	d.SetId(vm.Metadata.Name)
-	if d.Get("run_on_launch").(bool) {
-		diags, _ = waitForVirtualMachineToTargetState(ctx, d, d.Get("cluster_uid").(string), d.Get("name").(string), d.Get("namespace").(string), diags, c, "create", "Running")
-		if diags.HasError() {
-			return diags
+	var vm *models.V1ClusterVirtualMachine
```

Comment:
not a review, but a question - how do we currently support creating vms from templates?

Reply:
Currently, we are supporting the image template through image_url.